app/payment-link-v2/reference/get-index.controller.test.js

```diff
@@ -0,0 +1,109 @@
+'use strict'
+
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+const { expect } = require('chai')
+
+const productFixtures = require('../../../test/fixtures/product.fixtures')
+const serviceFixtures = require('../../../test/fixtures/service.fixtures')
+const Service = require('../../models/Service.class')
+const responseSpy = sinon.spy()
+const { NotFoundError } = require('../../errors')
+const Product = require('../../models/Product.class')
+
+const mockResponses = {
+  response: responseSpy
+}
+
+let req, res
+
+describe('Reference Page - GET controller', () => {
+  const mockCookie = {
+    getSessionVariable: sinon.stub()
+  }
+
+  const controller = proxyquire('./get-index.controller', {
+    '../../utils/response': mockResponses,
+    '../../utils/cookie': mockCookie
+  })
+
+  const service = new Service(serviceFixtures.validServiceResponse().getPlain())
+
+  beforeEach(() => {
+    mockCookie.getSessionVariable.reset()
+    responseSpy.resetHistory()
+  })
+
+  describe('when product.reference_enabled=true', () => {
+    const product = new Product(productFixtures.validCreateProductResponse({
+      type: 'ADHOC',
+      reference_enabled: true,
+      price: null
+    }).getPlain())
+
+    it('when the reference is NOT in the session, then it should display the reference page ' +
+      'and set the back link to the PRODUCT page', () => {
+      mockCookie.getSessionVariable.withArgs(req, 'referenceNumber').returns(null)
+
+      req = {
+        correlationId: '123',
+        product,
+        service
+      }
+      res = {}
+      controller(req, res)
+
+      expect(responseSpy.called).to.equal(true)
```

Comment:
It's better to use `sinon.assert.calledWith` etc. as this gives much nicer error messages when the assertion fails than using chai expect, as per https://pay-team-manual.cloudapps.digital/manual/coding-standards/node-js-testing-strategy.html#assertions

Reply:
Thanks - I have update all references using this.